### PR TITLE
Support round long function for decimal places < 0

### LINKED ISSUE #10025: Incorrect return value for ROUND_N with negative N for types other than DECIMAL
`ROUND(SMALLINT '99', -1)` should return 100, now it returns 99.

### PR DESCRIPTION
**Pull request by ebyhr**
Fix #10025

### CODE REVIEW

File: presto-main/src/test/java/com/facebook/presto/operator/scalar/TestMathFunctions.java [hunk 1]
```diff
@@ -763,14 +763,28 @@ public void testRound()
...
         assertFunction("round(-3, INTEGER '0')", INTEGER, -3);
+        assertFunction("round(99, INTEGER '-1')", INTEGER, 100);
+        assertFunction("round(-99, INTEGER '-1')", INTEGER, -100);
         assertFunction("round(BIGINT '3', 0)", BIGINT, 3L);
+        assertFunction("round(BIGINT '99', -1)", BIGINT, 100L);
+        assertFunction("round(BIGINT '-99', -1)", BIGINT, -100L);
         assertFunction("round( 3000000000, 0)", BIGINT, 3000000000L);
         assertFunction("round(-3000000000, 0)", BIGINT, -3000000000L);
+        assertFunction("round( 3999999999, -1)", BIGINT, 4000000000L);
+        assertFunction("round(-3999999999, -1)", BIGINT, -4000000000L);
```
Comment (findepi) [inline presto-main/src/test/java/com/facebook/presto/operator/scalar/TestMathFunctions.java]: add a cases
- `(9223372036854775807, -3)`
- `(9223372036854775807, -19)` (should overflow)
- `(9223372036854775807, -20)`
- `(9223372036854775807, -2147483648)`

File: presto-main/src/test/java/com/facebook/presto/operator/scalar/TestMathFunctions.java [hunk 2]
```diff
@@ -763,14 +763,28 @@ public void testRound()
...
         assertFunction("round(SMALLINT '3', 0)", SMALLINT, (short) 3);
+        assertFunction("round(SMALLINT '99', -1)", SMALLINT, (short) 100);
+        assertFunction("round(SMALLINT '-99', -1)", SMALLINT, (short) -100);
         assertFunction("round(3, 0)", INTEGER, 3);
         assertFunction("round(-3, 0)", INTEGER, -3);
+        assertFunction("round(99, -1)", INTEGER, 100);
+        assertFunction("round(-99, -1)", INTEGER, -100);
         assertFunction("round(-3, INTEGER '0')", INTEGER, -3);
+        assertFunction("round(99, INTEGER '-1')", INTEGER, 100);
+        assertFunction("round(-99, INTEGER '-1')", INTEGER, -100);
```
Comment (findepi) [inline presto-main/src/test/java/com/facebook/presto/operator/scalar/TestMathFunctions.java]: add case
* `(12355, -2) → 12400)`
* `(12345, -2) → 12300)`
* `(2147483647, -9) → 2000000000`
* `(2147483647, -10) → 0`
* `(2147483647, -100) → 0`
* `(2147483647, -2147483648) → 0`

**Comment by ebyhr** [comment]
I'm sorry but I close this PR because it seems it's not simpler than my expected and I don't have much time. If I find a better way to fix, I may reopen this PR though.

**Comment by findepi** [comment]
@ebyhr please keep this open. If you don't have the time to continue, someone may take this over.

**Comment by ebyhr** [comment]
@findepi I see :) Reopened.

**Pull request closed without merging**